### PR TITLE
Client: UX - Create Notifications and Header UI Controls (fixes #18)

### LINKED ISSUE #18: Client: UX - Make the menu dynamic
* Place a hamburger icon to the left of the Beans title.
* On click, change the menu's display css to flex
* Animate the entry of the menu

### PR DESCRIPTION
**Pull request by joshcox**
Brought App back to a working state; created Header and Notifications UI Control; ensured that original DOM markup was retained using the new classes; removed skeleton HTML that lived in beans.html (fixes #18)

### CODE REVIEW

**Comment by joshcox** [comment]
For some reason our app.js was throwing an error when running `npm start`, so I fixed that issue so you can start up the app now and see it @ http://localhost:3000

Also - I commented the BeansView stuff out for now and used our UI Controls to create the DOM. See the index.html file for that stuff. When we create the Content and BeansView UI Controls, that will get removed again. I just wanted to verify that our stuff was working as expected.

In addition to that, I did some updates on the Menu class that you'll want to check out. That was my bad - I've forgotten a lot of the jQuery syntax and I missed some of our mistakes.

File: lib/client/js/ui/controls/Notifications.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        // turn the arguments of this constructor into an array
+        var args = Array.prototype.slice.call(arguments);
+
+        /*
+         * @see {@link http://tobyho.com/2010/11/22/javascript-constructors-and/}
+         */
+
+        // Apply the Control constructor to the Notifications instance with the input arguments
+        Control.apply(this, args);
+    }
```
Comment (mjayjones13) [inline lib/client/js/ui/controls/Notifications.js:21]: Question: So, I've been noticing that constructors don't have a semicolon after their closing bracket. Yet, every other function seems to have one. Is this JS standard?

File: lib/client/js/ui/controls/Header.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        // Apply the Control constructor to the Header instance with the input arguments
+        Control.apply(this, args);
+    }
+
+    Header.prototype = new Control();
+
+    /**
+     * A "pure" view method, which is to say that mutation of the
+     * Menu instance should not happen here. It just takes what's in
```
Comment (mjayjones13) [inline lib/client/js/ui/controls/Header.js]: Minor: Header instance, not Menu instance

**Comment by mjayjones13** [comment]
Looks good to me other than that minor comment correction

File: lib/client/js/ui/controls/Notifications.js (hunk 1 above)
Comment (joshcox) [reply lib/client/js/ui/controls/Notifications.js:21]: There's a difference between a function declaration, `function foo() {}`, and an assignment of a variable to a function, `var foo = function () {};`. Function declarations should not have a semicolon. The first answer [here](http://stackoverflow.com/questions/1834642/why-should-i-use-a-semicolon-after-every-function-in-javascript) provides a good explanation as to why function declarations don't need them/don't get them.

File: lib/client/js/ui/controls/Header.js (hunk 2 above)
Comment (joshcox) [reply lib/client/js/ui/controls/Header.js]: This has been addressed!

**Pull request merged**